lib/pdk/util/ruby_version.rb

```diff
@@ -6,26 +6,62 @@ class RubyVersion
       class << self
         extend Forwardable
 
-        def_delegators :instance, :gem_path, :gem_home, :versions, :available_puppet_versions
+        def_delegators :instance, :gem_path, :gem_home, :available_puppet_versions
 
-        attr_writer :instance
+        attr_reader :instance
 
         def instance
-          @instance ||= new
+          if @instance.nil?
+            @instance = {}
+            @instance.default_proc = proc do |hash, key|
+              hash[key] = new(key)
+            end
+          end
+          @instance[active_ruby_version]
+        end
+
+        def active_ruby_version
+          @active_ruby_version || default_ruby_version
+        end
+
+        def use(version)
+          if versions.key?(version)
+            @active_ruby_version = version
+          else
+            raise ArgumentError, _('Unknown Ruby version "%{ruby_version}"') % {
+              ruby_version: version,
+            }
+          end
+        end
+
+        def scan_for_packaged_rubies
+          { '2.4.3' => '2.4.0' }
```

Comment:
The contents do not match the name of the function. What gives?

Reply:
+1

Reply:
Yeah, that was method was just moved around in the file, I was going to implement it in the next PR.